opentelemetry-sdk/src/opentelemetry/sdk/_metrics/_internal/__init__.py

```diff
@@ -373,12 +373,41 @@ def __init__(
     def force_flush(self, timeout_millis: float = 10_000) -> bool:
         deadline_ns = _time_ns() + timeout_millis * 10**6
 
+        metric_reader_error = {}
+
         for metric_reader in self._sdk_config.metric_readers:
             current_ts = _time_ns()
-            if current_ts >= deadline_ns:
-                raise Exception("Timed out while flushing metric readers")
-            metric_reader.collect(
-                timeout_millis=(deadline_ns - current_ts) / 10**6
+            try:
+                if current_ts >= deadline_ns:
+                    raise Exception("Timed out while flushing metric readers")
+                metric_reader.collect(
+                    timeout_millis=(deadline_ns - current_ts) / 10**6
+                )
+
+            # pylint: disable=broad-except
+            except Exception as error:
+
+                metric_reader_error[metric_reader] = error
+
+        if self._atexit_handler is not None:
+            unregister(self._atexit_handler)
+            self._atexit_handler = None
```

Comment:
This is force_flush right? Why are we doing this?

Reply:
My bad, removing.